### PR TITLE
change file tree

### PR DESCRIPTION
**Pull request by zhaochenyang20**
## Motivation

https://docs.google.com/document/d/1pPsLYOrMs6w31xfxx6DCftgUQmLqSSNMwAUehV5qNyg/edit?usp=sharing

As discussed, I changed the file tree and CI accordingly.

I use sub directories and created all the docs we will finish in the nearest future.

## Checklist

- [x] Format your code according to the [Contributor Guide](https://github.com/sgl-project/sglang/blob/main/docs/contributor_guide.md).
- [x] Add unit tests as outlined in the [Contributor Guide](https://github.com/sgl-project/sglang/blob/main/docs/contributor_guide.md).
- [x] Update documentation as needed, including docstrings or example tutorials.

### CODE REVIEW

**Review by merrymercy — COMMENTED** [review]
do not add new empty files.

File: .github/workflows/deploy-docs.yml [hunk 1]
```diff
@@ -38,7 +38,8 @@ jobs:
           GITHUB_TOKEN: ${{ secrets.PAT_TOKEN }}
         run: |
           cd docs
-          for nb in *.ipynb; do
+          rm -rf _build
+          find . -name "*.ipynb" | while read nb; do
```
Comment (merrymercy) [inline .github/workflows/deploy-docs.yml]: use `make compile` here?

File: docs/deploy.py
Comment (merrymercy) [inline docs/deploy.py:1]: Let us still keep this file for manual deployement without github action runners.

**Review by merrymercy — CHANGES_REQUESTED** [review]

File: docs/index.rst [hunk 2]
```diff
@@ -15,35 +15,35 @@ The core features include:
...
+   starts/send_request.ipynb
 
 
 .. toctree::
    :maxdepth: 1
    :caption: Backend Tutorial
 
-   openai_api.ipynb
-   backend.md
+   backend/openai_api.ipynb
```
Comment (merrymercy) [inline docs/index.rst:26]: do we also need to update these links? https://github.com/sgl-project/sglang/blob/b9fd178f1b7bab721b384c017dcec30a3ba0f323/README.md?plain=1#L42-L49

Comment (zhaochenyang20) [reply docs/index.rst:26]: Let me check this.

Comment (zhaochenyang20) [reply docs/index.rst:26]: Yeah. We need to do this @merrymercy

File: .github/workflows/deploy-docs.yml [hunk 3]
```diff
@@ -38,14 +38,8 @@ jobs:
...
           cd docs
-          for nb in *.ipynb; do
-            if [ -f "$nb" ]; then
-              echo "Executing $nb"
-              jupyter nbconvert --to notebook --execute --inplace "$nb" \
-                --ExecutePreprocessor.timeout=600 \
-                --ExecutePreprocessor.kernel_name=python3
-            fi
-          done
+          rm -rf _build
```
Comment (merrymercy) [inline .github/workflows/deploy-docs.yml:41]: make clean

File: .github/workflows/execute-notebook.yml [hunk 4]
```diff
@@ -44,11 +44,5 @@ jobs:
...
-          for nb in *.ipynb; do
-            if [ -f "$nb" ]; then
-              echo "Executing $nb"
-              jupyter nbconvert --to notebook --execute --inplace "$nb" \
-                --ExecutePreprocessor.timeout=600 \
-                --ExecutePreprocessor.kernel_name=python3
-            fi
-          done
\ No newline at end of file
+          rm -rf _build
```
Comment (merrymercy) [inline .github/workflows/execute-notebook.yml:47]: make clean

File: .github/workflows/deploy-docs.yml (hunk 3 above)
Comment (merrymercy) [inline .github/workflows/deploy-docs.yml]: make clean

File: .github/workflows/execute-notebook.yml (hunk 4 above)
Comment (merrymercy) [inline .github/workflows/execute-notebook.yml]: make clean

File: docs/deploy.py
Comment (merrymercy) [inline docs/deploy.py:1]: keep this file

**Review by merrymercy — CHANGES_REQUESTED** [review]
update README.md

**Pull request merged**